### PR TITLE
Use sorted imports + enable more linters

### PR DESCRIPTION
**Pull request by tisnik**
## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Configuration Update

## Description

- Use sorted imports everywhere
- More linters can be enabled right now

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- See CI jobs

### CODE REVIEW

File: app/models/models.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
-from pydantic import BaseModel
 from typing import Union
 
+from pydantic import BaseModel
```
Comment (thoraxe) [inline app/models/models.py:3]: t comes after p -- this isn't sorted?

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.ruff]
 
-select = ["E", "F", "W", "C", "S"]
+select = ["E", "F", "W", "C", "Q", "SLF", "I"]
```
Comment (thoraxe) [inline pyproject.toml:3]: What does this line do?

File: src/docs/docs_summarizer.py [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import os
+
 import llama_index
```
Comment (thoraxe) [inline src/docs/docs_summarizer.py:3]: why is this still here (l) if we are alpha sorting, and then you left the other llama-related a few lines down?

**Review by thoraxe — CHANGES_REQUESTED** [review]
I don't fully understand this PR. In some cases, it appears that you are not alpha sorting, but rather some other ordering comes first and then, within that ordering, you are alpha sorting.

Is there some PEP that describes the correct way to sort and group imports?

File: app/models/models.py (hunk 1 above)
Comment (tisnik) [reply app/models/models.py:3]: it sorts, but after empty line, basically the new "import group" is sorted on its own. It's a bit similar like in Go

File: pyproject.toml (hunk 2 above)
Comment (tisnik) [reply pyproject.toml:3]: It enables more linters. All of them can be listed by using `ruff linter`. Q - check string quotes, SLF - self. param check, I - import sorting

**Comment by tisnik** [comment]
> I don't fully understand this PR. In some cases, it appears that you are not alpha sorting, but rather some other ordering comes first and then, within that ordering, you are alpha sorting.
> 
> Is there some PEP that describes the correct way to sort and group imports?

Ideal behavior is:

1) sort default packages
2) sort third party packages
3) sort in-project packages

File: src/docs/docs_summarizer.py (hunk 3 above)
Comment (tisnik) [reply src/docs/docs_summarizer.py:3]: please look below:
Ideal behavior is:

    "future" imports first
    sort default packages
    sort third party packages
    sort in-project packages

File: pyproject.toml (hunk 2 above)
Comment (thoraxe) [reply pyproject.toml]: Can you make the linter change a separate PR? I want Ben and others to discuss that.

File: src/docs/docs_summarizer.py (hunk 3 above)
Comment (thoraxe) [reply src/docs/docs_summarizer.py:3]: :+1:

File: pyproject.toml (hunk 2 above)
Comment (tisnik) [reply pyproject.toml]: yes sure, I'll delete this commit ASAP

Comment (tisnik) [reply pyproject.toml]: done

**Pull request merged**